### ISSUE TITLE
You can autopkg repo-add file:///some/where that can't be autopkg repo-delete'd later

### ISSUE THREAD

**Issue by KWik** [autopkg/autopkg#703]
**Description of the problem**
Just for fun I did autopkg repo-add file:///Users/klausw/stage/autopkg-recipes to test something after someone asked in Slack about adding a local repo. And it worked. I could add it. But I did not want to have that in my path, so I tried deleting it. Making this ticket I re-added it and it works really well:

```
IFIKK-MacBook-Pro:stage klausw$ autopkg repo-add file:///Users/klausw/stage/autopkg-recipes
Attempting git pull...
Already up to date.

Adding /Users/klausw/Library/AutoPkg/RecipeRepos/.Users.klausw.stage.autopkg-recipes to RECIPE_SEARCH_DIRS...
```

But later, repo-update and repo-delete fails.

```
IFIKK-MacBook-Pro:stage klausw$ autopkg repo-update file:///Users/klausw/stage/autopkg-recipes
ERROR: Can't find an installed repo for file:///Users/klausw/stage/autopkg-recipes
IFIKK-MacBook-Pro:stage klausw$ autopkg repo-delete file:///Users/klausw/stage/autopkg-recipes
ERROR: Can't find an installed repo for file:///Users/klausw/stage/autopkg-recipes
```

How repo-list shows it:
IFIKK-MacBook-Pro:stage klausw$ autopkg repo-list
```
/Users/klausw/Library/AutoPkg/RecipeRepos/.Users.klausw.stage.autopkg-recipes (file:///Users/klausw/stage/autopkg-recipes)
/Users/klausw/Library/AutoPkg/RecipeRepos/com.github.KWik.autopkg-recipes (https://github.com/KWik/autopkg-recipes.git)
/Users/klausw/Library/AutoPkg/RecipeRepos/com.github.attak1t.attak1t-recipes (https://github.com/attak1t/attak1t-recipes.git)
```

**Expected behavior**
Expected behavior should of course be that if I can add it, and adding it sees it as up to date, then I should be able to update and delete it as well.

**Workaround**
PlistBuddy to the rescue.

**Version (please complete the following information):**
 - OS version: 10.14.6
- AutoPkg Version: 2.2

**Comment by homebysix** [comment]
Hi @KWik - I can reproduce this issue. However, is there a specific reason not to just add the local repository to the `RECIPE_SEARCH_DIRS` preference instead of using `repo-add`?

Presumably either the local repo is a clone of a hosted remote (in which case the remote URL should be used with `repo-add`) or the repo is not synced remotely and should be treated as a normal folder full of recipes (by appending to the search path).

I'd be more inclined to warn and skip any file:// URLs rather than try to support them as quasi-remotes.

**Comment by nmcspadden** [comment]
I agree with @homebysix here - autopkg should warn you not to use `file://` URLs.

**Comment by nmcspadden** [comment]
Fixed in 326348f38d58710313ae6e62a3fbca3da0d55447.
